docs/src/architecture/domain-model.md

```diff
@@ -0,0 +1,121 @@
+  * **Events**:
+    * contain other real-time information from the sensor network, such as the triggering of a door sensor
+    * an **event** is triggered by a **data point** (e.g. when it changes to a specific value as `0` or `1`), and contains an optional message text.
+  * **Alarms**:
+    * similar to Events, but the user or operator of the system has to take action to resolve the alarm
+    * an **alarm** is triggered by a **data point** (e.g. when it changes to a specific value as `0` or `1`), and contains an optional alarm text.
+
+## thin-edge plugin concept
+
+### Plugin
+
+**thin-edge** realizes high level cloud functionality (e.g. _configuration management_ or _log management_) with **plugins**.
+  * a **plugin** accesses _ressources_ and _services_ of the device, as e.g.
+      * the device's _package manager_ is accessed by _software management_
+      * device's _configuration files_ are accessed by _configuration management_
+  * a **plugin** can be an (external) executable (e.g. as the `c8y_configuration_plugin` for _configuration management_)
+    or a thin-edge built-in software component (e.g. as for _software management_)
+  * usually a **plugin** runs on the **main-device**; thus it can access the _resources_ of the **main-device** directly
+  * to access _ressources_ of an **external child-device** a **plugin** needs another component, referred as **child-device agent**
+
+### Child-Device Agent
+  * a **child-device agent** is the counterpart of a **plugin**, that takes the responsibility to access to the **external child-device's** _resources_
+  * a **child-device agent** can serve one or more **plugins**
+  * a **child-device agent** can be installed and executed on the **external child-device**, or on the **main-device**
+    * if it runs in the **external child-device** it can access the _resources_ directly
+    * if it runs on the **main-device** it can use any (low-level) interfaces the **external child-device** provides to access those resources
+      * One main reason to install the **child-device agent** on the **main-device** is, when the **external child-device** cannot or shall not be altered.
+
+### Plugin-Contract
+
+a **plugin** defines and implements a specific **contract** for all interactions with a **child-device agent**
+  * part of the **contract** could be e.g.:
+      * the **child-device agent** must listen and react to certain requests of the **plugin**, e.g. on MQTT
+      * the **child-device agent** must provide/consume files to/from the **plugin** on purpose, e.g. via HTTP
+      * ...and more...
+  * a **plugin's** **contract** can be denoted with a unique name (e.g. `tedge_config`)
+    * based on that unique name a **child-device agent** can report and find **plugins** the child-device intends to contact (e.g. during providioning phase)
+    * those information can be provided to the cloud and other applications on the device site
```

Comment:
I din't quite understand what you were trying to communicate with that sentence.

Reply:
@albinsuresh , sorry, I am not sure which sentence you mean exactly. Can you please quote it?